### PR TITLE
RAB-1422: Spike to test FlySystem object delete on large client database

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

This is a spike to test the performance when we completely delete the catalog storage

**Result**
Average time : 0.091 seconds

### CODE REVIEW

**Comment by pierrejolly** [comment]
Test with 19708189 files => 0.12428689002991 sec

File: src/Akeneo/Platform/Installer/back/src/Infrastructure/Symfony/Resources/config/services.yml [hunk 1]
```diff
@@ -40,3 +40,7 @@ services:
       - '@database_connection'
     tags:
       - {name: akeneo.installer.user_configuration_resetter}
+
+  Akeneo\Platform\Installer\Infrastructure\StoragePurger\StoragePurger:
+    arguments:
+      - '@oneup_flysystem.catalog_storage_filesystem'
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Installer/back/src/Infrastructure/Symfony/Resources/config/services.yml:46]: You delete catalog storage that why it do not take time on madeira all of her file are in asset manager they didn't use the catalog storage (file/image attribute type)

File: src/Akeneo/Platform/Installer/back/src/Infrastructure/StoragePurger/StoragePurger.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    public function purgeStorage(FilesystemOperator $storage): void
+    {
+        foreach ($storage->listContents('.') as $content) {
+            if ($content->isDir()) {
+                $storage->deleteDirectory($content->path());
+            } else {
+                $storage->delete($content->path());
+            }
+        }
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Installer/back/src/Infrastructure/StoragePurger/StoragePurger.php:35]: In my opinion we can directly delete the `./` directory

```suggestion
       $storage->deleteDirectory('./');
```

**Review by StevenVAIDIE — COMMENTED** [review]
If we want to merge it like this we need to iterate on Flysystems

**Pull request closed without merging**